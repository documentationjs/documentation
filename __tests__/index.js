const documentation = require('../src/');
const os = require('os');
const path = require('path');
const fs = require('fs');

function inputs(contents) {
  const dirEntry = os.tmpdir();
  const paths = {};
  for (const filename in contents) {
    paths[filename] = path.join(dirEntry, '/', filename);
    fs.writeFileSync(paths[filename], contents[filename]);
  }
  return {
    paths
  };
}

function cleanup(comments) {
  comments.forEach(c => {
    delete c.context;
  });
}

test('lint', async function () {
  const { paths } = inputs({
    'index.js': '/** hi */var name = 1;'
  });

  const data = await documentation.lint([paths['index.js']], {});
  expect(data).toEqual('');
});

test('build', async function () {
  const { paths } = inputs({
    'index.js': '/** hi */var name = 1;'
  });

  const data = await documentation.build([paths['index.js']], {});
  cleanup(data);
  expect(data).toMatchSnapshot();

  const md = await documentation.formats.md(data);
  expect(md).toMatchSnapshot();

  const json = await documentation.formats.json(data, {});
  expect(json).toMatchSnapshot();
});

test('expandInputs', async function () {
  const { paths } = inputs({
    'index.js': '/** hi */var name = 1;'
  });

  {
    const data = await documentation.expandInputs([paths['index.js']], {
      parseExtension: ['js']
    });
    expect(data.length).toEqual(1);
  }

  {
    const data = await documentation.expandInputs([paths['index.js']], {
      parseExtension: ['js'],
      shallow: true
    });
    expect(data.length).toEqual(1);
  }
});
