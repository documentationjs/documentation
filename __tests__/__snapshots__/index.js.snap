// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`build 1`] = `
Array [
  Object {
    "augments": Array [],
    "description": Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "type": "text",
              "value": "hi",
            },
          ],
          "type": "paragraph",
        },
      ],
      "type": "root",
    },
    "errors": Array [],
    "examples": Array [],
    "implements": Array [],
    "loc": SourceLocation {
      "end": Position {
        "column": 9,
        "index": 9,
        "line": 1,
      },
      "filename": undefined,
      "identifierName": undefined,
      "start": Position {
        "column": 0,
        "index": 0,
        "line": 1,
      },
    },
    "members": Object {
      "events": Array [],
      "global": Array [],
      "inner": Array [],
      "instance": Array [],
      "static": Array [],
    },
    "name": "name",
    "namespace": "name",
    "params": Array [],
    "path": Array [
      Object {
        "kind": undefined,
        "name": "name",
      },
    ],
    "properties": Array [],
    "returns": Array [],
    "sees": Array [],
    "tags": Array [],
    "throws": Array [],
    "todos": Array [],
    "yields": Array [],
  },
]
`;

exports[`build 2`] = `
"<!-- Generated by documentation.js. Update this documentation by updating the source code. -->

## name

hi
"
`;

exports[`build 3`] = `
"[
  {
    \\"description\\": {
      \\"type\\": \\"root\\",
      \\"children\\": [
        {
          \\"type\\": \\"paragraph\\",
          \\"children\\": [
            {
              \\"type\\": \\"text\\",
              \\"value\\": \\"hi\\"
            }
          ]
        }
      ]
    },
    \\"tags\\": [],
    \\"loc\\": {
      \\"start\\": {
        \\"line\\": 1,
        \\"column\\": 0,
        \\"index\\": 0
      },
      \\"end\\": {
        \\"line\\": 1,
        \\"column\\": 9,
        \\"index\\": 9
      }
    },
    \\"augments\\": [],
    \\"examples\\": [],
    \\"implements\\": [],
    \\"params\\": [],
    \\"properties\\": [],
    \\"returns\\": [],
    \\"sees\\": [],
    \\"throws\\": [],
    \\"todos\\": [],
    \\"yields\\": [],
    \\"name\\": \\"name\\",
    \\"members\\": {
      \\"global\\": [],
      \\"inner\\": [],
      \\"instance\\": [],
      \\"events\\": [],
      \\"static\\": []
    },
    \\"path\\": [
      {
        \\"name\\": \\"name\\"
      }
    ],
    \\"namespace\\": \\"name\\"
  }
]"
`;
